cli/command/cli.go

```diff
@@ -126,6 +127,10 @@ func ShowHelp(err io.Writer) func(*cobra.Command, []string) error {
 
 // ConfigFile returns the ConfigFile
 func (cli *DockerCli) ConfigFile() *configfile.ConfigFile {
+	if cli.configFile != nil {
+		return cli.configFile
+	}
+	cli.configFile = cliconfig.LoadDefaultConfigFile(cli.err)
```

Comment:
Actually, looking at this; I wonder if this loads the correct config-file, because it's not taking the `--config` flag into account? 🤔 https://github.com/docker/cli/blob/0f337f1dfe574eb12eab8bb102a24f714cc79d86/cli/command/cli.go#L202-L210